magicctapipe/io/io.py

```diff
@@ -1419,3 +1420,132 @@ def resource_file(filename):
         Absolute path of the input filename
     """
     return files("magicctapipe").joinpath("resources", filename)
+
+
+def find_offset(data_magic_, data_lst_, N_start=0, N_end=20, initial_time_offset=None):
+    """
+    Find the time offset between MAGIC and LST.
+
+    Parameters
+    ----------
+    data_magic_ : str
+        Input filename for MAGIC
+    data_lst_ : str
+        Input filename for LST
+    N_start, N_end : int
+        Start and stop event number for MAGIC event
+    initial_time_offset : float
+        The initial time offset value for search the time offset
+
+    Returns
+    -------
+    t_magic_ave : float
+        The average the MAGIC timestamp
+    event_id_magic_ave : float
+        The average of the MAGIC event ID
+    time_offset_best : float
+        The best time offset found from the combination
+    n_coincident : float
+        The coincidence event number
+    """
+    if initial_time_offset is None:
+        time_offset_width1, time_offset_width2 = -4, 4
+        print(data_magic_["trigger_time"])
+        print(N_start)
+        time_select = (
+            data_lst_["trigger_time"].values
+            > (data_magic_["trigger_time"].values[N_start] + time_offset_width1)
+        ) & (
+            data_lst_["trigger_time"].values
+            < (data_magic_["trigger_time"].values[N_end] + time_offset_width2)
+        )
+        select = data_lst_["event_type"] == 32
```

Comment:
thise select is the same for both settings of the main if, so you can calculate it before if